builtin/providers/google/resource_compute_backend_service.go

```diff
@@ -0,0 +1,410 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func resourceComputeBackendService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeBackendServiceCreate,
+		Read:   resourceComputeBackendServiceRead,
+		Update: resourceComputeBackendServiceUpdate,
+		Delete: resourceComputeBackendServiceDelete,
+
+		Schema: map[string]*schema.Schema{
+			"backend": &schema.Schema{
+				Type: schema.TypeSet,
```

Comment:
I'm almost thinking that `TypeList` could be renamed to `TypeStruct` since that's the only safe way I'd use it - singletons.

It's now being used in a few other places, mostly computed lists, e.g. `aws_route53_zone.name_servers`, but I'm not quite convinced it's a good idea.

You may argue that APIs should just be returning items in a predictable order, but ... :pensive: